src/service.js

```diff
@@ -324,17 +383,29 @@ module.exports = function(mixinOptions) {
 					subscribe: filter
 						? withFilter(
 								() => this.pubsub.asyncIterator(tags),
-								async (payload, params, { ctx }) =>
-									payload !== undefined
-										? ctx.call(filter, { ...params, payload })
-										: false
+								async (payload, params) => {
+									return payload !== undefined
+										? this.createAsyncIteratorContext().call(filter, {
+												...params,
+												payload,
+										  })
+										: false;
+								}
 						  )
 						: () => this.pubsub.asyncIterator(tags),
-					resolve: (payload, params, { ctx }) =>
-						ctx.call(actionName, { ...params, payload }),
+					resolve: (payload, params) => {
+						return this.createAsyncIteratorContext().call(actionName, {
+							...params,
+							payload,
+						});
+					},
 				};
 			},
 
+			createAsyncIteratorContext() {
+				return this.broker.ContextFactory.create(this.broker, null, {}, {});
```

Comment:
This change seems to imply that a fresh moleculer context is not being created earlier (previous comment), but is instead necessary to be created here.  Is that correct?

If so, I have a few concerns.  First, it does seem like the code in the server setup area is creating several instances of Context that it really cannot make use of.  If they were being used then that would make sense, but if they're not it seems like a drag on connection/request time otherwise.  If they are being used, then I'm not sure I'm understanding the reason to create a new Context here.  Can you clarify for me?

Reply:
My other concern here is that if we need to create a fresh moleculer context here, at action call time (and it can't be done in the server setup callback) then we probably have a (likely preexisting) issue with request to the DataLoader instances.  If those instances are not being created for each new "request" (subscription, query, or mutation) then we're likely to run into caching issues on those instances.  That is, repeated requests several hours later will still be using the same DataLoader cached results, which is not the prescribed "DataLoader per request" pattern.